python/rmm/_lib/memory_resource.pyx

```diff
@@ -25,6 +25,7 @@ from libc.stddef cimport size_t
 from libc.stdint cimport int8_t, int64_t, uintptr_t
 from libcpp cimport bool
 from libcpp.memory cimport make_unique, unique_ptr
+from libcpp.optional cimport make_optional, optional
```

Comment:
@miscco I pushed some changes here -- now that we're using the STL version, we don't need the `extern` definition from Thrust at all.

Reply:
😻